spyder/widgets/editor.py

```diff
@@ -941,11 +941,33 @@ def __modified_readonly_title(self, title, is_modified, is_readonly):
             title = "(%s)" % title
         return title
 
-    def get_tab_text(self, filename, is_modified=None, is_readonly=None):
+    def get_tab_text(self, index, filename, is_modified=None, is_readonly=None):
         """Return tab title"""
-        return self.__modified_readonly_title(osp.basename(filename),
+        fname = self.get_file_title(index)
+        return self.__modified_readonly_title(fname,
                                               is_modified, is_readonly)
 
+    def get_file_title(self, index):
+        """Get tab title without ambiguation."""
+        finfo = self.data[index]
+        fname = osp.basename(finfo.filename)
+        for findex, source in enumerate(self.data):
+            fname_to_compare = osp.basename(source.filename)
+            if findex is not index and fname == fname_to_compare:
+                differ_path = sourcecode.differentiate_prefix(
+                                   sourcecode.path_components(finfo.filename), 
+                                   sourcecode.path_components(source.filename))
+                differ_path_length = len(differ_path)
+                if (differ_path_length > 20):

```

Comment:
If the length changes to 4 this will be the way the titles will be shown:

![image](https://cloud.githubusercontent.com/assets/16781833/20846760/204a71da-b899-11e6-9f7f-4d91e7a522dc.png)

Is this format ok?